projects/packages/forms/src/dashboard/inbox/index.js

```diff
@@ -1,20 +1,104 @@
+import { __, _n, sprintf } from '@wordpress/i18n';
+import { find, includes, map } from 'lodash';
+import { useDispatch, useSelector } from 'react-redux';
 import Layout from '../components/layout';
+import {
+	fetchResponses,
+	getResponses,
+	getTotalResponses,
+	isFetchingResponses,
+} from '../state/inbox';
 import InboxList from './list';
 import InboxResponse from './response';
 
 import './style.scss';
 
+const RESPONSES_PER_PAGE = 10;
+
 const Inbox = () => {
+	const [ currentResponseId, setCurrentResponseId ] = useState( -1 );
+	// const [ showResponseView, setShowResponseView ] = useState( false );
+	const [ searchText, setSearchText ] = useState( '' );
+
+	const dispatch = useDispatch();
+
+	const [ loading, responses, total ] = useSelector( state => {
+		return [ isFetchingResponses( state ), getResponses( state ), getTotalResponses( state ) ];
+	} );
+
+	useEffect( () => {
+		fetchResponses( {}, RESPONSES_PER_PAGE )( dispatch );
+	}, [ dispatch ] );
+
+	useEffect( () => {
+		if ( responses.length === 0 || includes( map( responses, 'id' ), currentResponseId ) ) {
+			return;
+		}
+
+		setCurrentResponseId( responses[ 0 ].id );
+	}, [ responses, currentResponseId ] );
+
+	const handleLoadMore = useCallback( () => {
+		fetchResponses( { search: searchText }, RESPONSES_PER_PAGE, responses.length )( dispatch );
+	}, [ searchText, responses, dispatch ] );
+
+	const handleSearch = useCallback(
+		event => {
+			event.preventDefault();
+			fetchResponses( { search: searchText }, RESPONSES_PER_PAGE )( dispatch );
+		},
+		[ searchText, dispatch ]
+	);
+
+	const numberOfResponses = sprintf(
+		/* translators: %s: Number of responses. */
+		_n( '%s response', '%s responses', total, 'jetpack-forms' ),
+		total
+	);
+
 	return (
-		<Layout title={ __( 'Responses', 'jetpack-forms' ) }>
+		<Layout title={ __( 'Responses', 'jetpack-forms' ) } numberOfResponses={ numberOfResponses }>
+			<div className="jp-forms__actions">
+				<form className="jp-forms__actions-form">
+					<SelectControl
+						options={ [
+							{ label: __( 'Bulk actions', 'jetpack-forms' ), value: '' },
+							{ label: __( 'Trash', 'jetpack-forms' ), value: 'trash' },
+							{ label: __( 'Move to spam', 'jetpack-forms' ), value: 'spam' },
+						] }
+					/>
+					<Button variant="secondary">{ __( 'Apply', 'jetpack-forms' ) }</Button>
+				</form>
+				<form className="jp-forms__actions-form" onSubmit={ handleSearch }>
+					<InputControl onChange={ setSearchText } value={ searchText } />
+					<Button type="submit" variant="secondary">
+						{ __( 'Search', 'jetpack-forms' ) }
+					</Button>
+				</form>
+			</div>
 			<div className="jp-forms__inbox-content">
 				<div className="jp-forms__inbox-content-column">
-					<InboxList />
+					<InboxList
+						currentResponseId={ currentResponseId }
+						hasMore={ responses.length < total }
+						loading={ loading }
+						onLoadMore={ handleLoadMore }
+						onSelectionChange={ setCurrentResponseId }
```

Comment:
Ok, I see the issue now. `onSelectionChange` is a handler for when you toggle the table checkboxes. When it comes to selecting individual feedback items, that's actually not implemented yet, as we haven't decided if we're using links or doing internal state manipulation.
In the long run, we definitely want URLs but feel free to add another handler if you need something for now.

Reply:
Yes, I picked up on this as well too late into the change, maybe some "active" state for the rows will help, with its own `onActiveChange`, etc handlers. I'll see if I can make some changes here, but I'm thinking this is already too big and, as well as with other things, I think we can move on to further iterations.

Reply:
That's fine but in that case I'd prefer to leave the `<Table />` component as even if it means we'd temporarily loose the ability to select the active row. Like you said, it doesn't work correctly right now either so let's just avoid the confusion and implement it the way it's supposed to be in the next step.